src/utils/Event.ts

```diff
@@ -0,0 +1,50 @@
+class Event<T, R = void> {
```

Comment:
Generic event and async event class. This is used by the new coderunner. Ultimately, a glorified observer pattern.